worlds/inscryption/__init__.py

```diff
@@ -0,0 +1,126 @@
+from .Options import InscryptionOptions, Goal, EpitaphPiecesRandomization, PaintingChecksBalancing
+from .Items import act1_items, act2_items, act3_items, filler_items, base_id, InscryptionItem, ItemDict
+from .Locations import act1_locations, act2_locations, act3_locations, regions_to_locations
+from .Regions import inscryption_regions_all, inscryption_regions_act_1
+from typing import Dict, Any
+from . import Rules
+from BaseClasses import Region, Item, Tutorial, ItemClassification, LocationProgressType
+from worlds.AutoWorld import World, WebWorld
+
+
+class InscrypWeb(WebWorld):
+    theme = "dirt"
+
+    guide_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Inscryption Archipelago Multiworld",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["DrBibop"]
+    )
+
+    guide_fr = Tutorial(
+        "Multiworld Setup Guide",
+        "Un guide pour configurer Inscryption Archipelago Multiworld",
+        "Français",
+        "setup_fr.md",
+        "setup/fr",
+        ["Glowbuzz"]
+    )
+
+    tutorials = [guide_en, guide_fr]
+
+    bug_report_page = "https://github.com/DrBibop/Archipelago_Inscryption/issues"
+
+
+class InscryptionWorld(World):
+    """
+    Inscryption is an inky black card-based odyssey that blends the deckbuilding roguelike,
+    escape-room style puzzles, and psychological horror into a blood-laced smoothie.
+    Darker still are the secrets inscrybed upon the cards...
+    """
+    game = "Inscryption"
+    web = InscrypWeb()
+    options_dataclass = InscryptionOptions
+    options: InscryptionOptions
+    all_items = act1_items + act2_items + act3_items + filler_items
+    item_name_to_id = {item['name']: i + base_id for i, item in enumerate(all_items)}
+    all_locations = act1_locations + act2_locations + act3_locations
+    location_name_to_id = {location: i + base_id for i, location in enumerate(all_locations)}
+
+    def create_item(self, name: str) -> Item:
+        item_id = self.item_name_to_id[name]
+        item_data = self.all_items[item_id - base_id]
+        return InscryptionItem(name, item_data['classification'], item_id, self.player)
+
+    def create_items(self) -> None:
+        nb_items_added = 0
+        useful_items = (act1_items + act2_items + act3_items) if self.options.goal != Goal.option_first_act \
+            else act1_items
+
+        if self.options.goal != Goal.option_first_act:
```

Comment:
I don't love the hard-coded numbers here, but like, it works 🤷

Reply:
wdym by hard-coded numbers? Do you have an alternate solution?

Reply:
The lines like `useful_items[len(act1_items) + 2]` where it's taking advantage of the very particular order/structure of the items instead of searching by name. It's just easier to mess up, especially if you ever change anything in the tables. I don't think it _needs_ to change, but figured I should mention it

Reply:
To be fair, this stems from my lack of Python experience. What would be the proper way to search by name, then? It's a list of ItemDict which contains a name, count and classification.

Reply:
I'll be honest, I was trying to think of better ways, but with the way it's structured, I honestly think other ways might just be worse. So how it is seems fine, and it's mostly personal preference, AP has a good chunk of magic numbers in its worlds and this one is honestly quite tame.